### PR TITLE
SAMZA-2301: Add non-null checks in JobModel read control-flow in standalone.

### PR DESCRIPTION
**Pull request by shanthoosh**

### CODE REVIEW

File: samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java [hunk 1]
```diff
@@ -403,12 +406,14 @@ private GrouperMetadataImpl getGrouperMetadata(String jobModelVersion, List<Proc
...
-          taskToProcessorId.put(taskModel.getTaskName(), containerModel.getId());
-          for (SystemStreamPartition partition : taskModel.getSystemStreamPartitions()) {
-            taskToSSPs.computeIfAbsent(taskModel.getTaskName(), k -> new ArrayList<>());
-            taskToSSPs.get(taskModel.getTaskName()).add(partition);
+      if (jobModel != null) {
+        for (ContainerModel containerModel : jobModel.getContainers().values()) {
+          for (TaskModel taskModel : containerModel.getTasks().values()) {
+            taskToProcessorId.put(taskModel.getTaskName(), containerModel.getId());
+            for (SystemStreamPartition partition : taskModel.getSystemStreamPartitions()) {
+              taskToSSPs.computeIfAbsent(taskModel.getTaskName(), k -> new ArrayList<>());
```
Comment (rmatharu) [inline samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java]: Can replace with 
`taskToSSPs.putIfAbsent(taskModel.getTaskName(), new ArrayList<>());` ?

File: samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java [hunk 2]
```diff
@@ -538,14 +546,15 @@ public ZkJobModelVersionChangeHandler(ZkUtils zkUtils) {
...
+          jobModelVersion = (String) data;
+
 
           LOG.info("Got a notification for new JobModel version. Path = {} Version = {}", dataPath, data);
 
           newJobModel = readJobModelFromMetadataStore(jobModelVersion);
           LOG.info("pid=" + processorId + ": new JobModel is available. Version =" + jobModelVersion + "; JobModel = " + newJobModel);
 
-          if (!newJobModel.getContainers().containsKey(processorId)) {
+          if (newJobModel != null && !newJobModel.getContainers().containsKey(processorId)) {
```
Comment (mynameborat) [inline samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java]: Until this change, the else code path corresponds to the case where the processor is ensured to be part of the new job model generated. However, with this change, we blur the distinction between processor being part of the job model vs job model unavailable. 
I suppose the by product of that is you can potentially have a stream processor running that is not part of the quorum or doesn't have any work to do. 
If that is intended, please add a comment and link the existing ticket.

**Comment by prateekm** [comment]
@shanthoosh Is this ready to be merged?

**Comment by shanthoosh** [comment]
Yes, @prateekm . This branch had a bunch of conflicts. Will open a new request in the future with the merge-conflicts resolved.

**Pull request closed without merging**